compact_str/src/lib.rs

```diff
@@ -1191,10 +1191,10 @@ impl CompactString {
         }
 
         // Note: we don't use collect::<Result<_, _>>() because that fails to pre-allocate a buffer,
-        // even though the size of our iterator, `buf`, is known ahead of time.
+        // even though the size of our iterator, `v`, is known ahead of time.
         //
         // rustlang issue #48994 is tracking the fix
-        let mut result = CompactString::with_capacity(v.len());
+        let mut result = CompactString::with_capacity(v.len() / 2);
```

Comment:
I forgot the `/ 2` in my last PR.

* `v.len() / 2` assumes that most of the input is ASCII.
* `v.len()` assumes that most of the input non-western European scripts in the [BMP](https://codepoints.net/basic_multilingual_plane), e.g. Korean, Japanese, Chinese, Russian …
* `v.len() / 2 * 3` assumes that most of input is text in uncommon or dead scripts
* `v.len() * 2` assumes that most of the input is emojis 💩

Reply:
Good call out, thanks for the update!